mautic/mautic
# Custom Field Alias constraint for special keywords added

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5694
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The problem is that if a user creates new custom field with alias of `hit_url_count` for example and then she'll create new segment filter for this `hit_url_count` it won't work with the values of that custom field but with the page hit counts. This may be confusing and even cause bugs. This PR adds validation which doesn't allow user to save a custom field with such keyword.
![screen shot 2018-03-20 at 16 54 05](https://user-images.githubusercontent.com/1235442/37668337-8648fcb4-2c64-11e8-88ef-82e53814e63f.png)


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Try to create new custom field with alias of `sessions` or `hit_url_count`. It will let you do that. Bad!

#### Steps to test this PR:
1. Checkout this PR
2. Clear cache
3. Delete the CF you created during testing.
4. Try to create it again. It will show you a validation error message as in the screenshot above. Good!
5. Try to save with another alias to ensure that's not broken.
